### PR TITLE
Add support for Event dependent configuration and changing concurrency to parallel

### PR DESCRIPTION
**Pull request by yoelcabo**

### CODE REVIEW

File: lib/logstash/outputs/CiscoZeus.rb [hunk 1]
```diff
@@ -22,9 +22,11 @@ def register
   end # def register
 
   def multi_receive(events)
-    result = @zeus_client.send_logs(@log_name, events)
-    if not result.success?
-      STDERR.puts "Failed to send data to zeus: " + result.data.to_s
+    events.group_by{ |ev| ev.sprintf(@log_name) }.each do |log_name, events_group| 
+      result = @zeus_client.send_logs(log_name, events_group)
+      if not result.success?
```
Comment (kosukemori) [inline lib/logstash/outputs/CiscoZeus.rb:27]: You can use `unless` instead of `if not`

File: lib/logstash/outputs/CiscoZeusMetrics.rb [hunk 2]
```diff
@@ -29,10 +29,12 @@ def register
...
 
   def multi_receive(events)
-    metrics = events.map{|event| reform(event)}
-    result = @zeus_client.send_metrics(@metric_name, metrics)
-    if not result.success?
-      STDERR.puts "Failed to send data to zeus: " + result.data.to_s
+    events.group_by{ |ev| ev.sprintf(@metric_name) }.each do |metric_name, events_group|
+      metrics = events_group.map{|event| reform(event)}
+      result = @zeus_client.send_metrics(metric_name, metrics)
+      if not result.success?
```
Comment (kosukemori) [inline lib/logstash/outputs/CiscoZeusMetrics.rb:35]: same as above

File: lib/logstash/outputs/CiscoZeus.rb (hunk 1 above)
Comment (yoelcabo) [reply lib/logstash/outputs/CiscoZeus.rb:27]: ohh nice! ruby is beautiful!
But I think in this case `if not` is more readable. "If the result is not successful print to stderr" sounds more logical than "Unless the result is successful print to stderr"

**Comment by victorfonsec4** [comment]
LGTM

**Pull request merged**